source/extensions/common/tap/tap_config_base.cc

```diff
@@ -13,22 +13,29 @@ namespace Extensions {
 namespace Common {
 namespace Tap {
 
-void Utility::addBufferToProtoBytes(envoy::data::tap::v2alpha::Body& output_bytes,
-                                    uint32_t max_buffered_bytes, const Buffer::Instance& data) {
+bool Utility::addBufferToProtoBytes(envoy::data::tap::v2alpha::Body& output_bytes,
+                                    uint32_t max_buffered_bytes, const Buffer::Instance& data,
+                                    uint32_t buffer_start_offset, uint32_t buffer_length_to_copy) {
   // TODO(mattklein123): Figure out if we can use the buffer API here directly in some way. This is
   // is not trivial if we want to avoid extra copies since we end up appending to the existing
   // protobuf string.
+  ASSERT(buffer_start_offset + buffer_length_to_copy <= data.length());
+  const uint32_t final_bytes_to_copy =
+      std::min(max_buffered_bytes, buffer_length_to_copy - buffer_start_offset);
```

Comment:
Let's say we have `data` length 10, `buffer_length_to_copy` is 4 and `buffer_start_offset` is 6. Won't the second term in `std::min` be negative?

Reply:
Yes, I think I broke this when I changed this for @lizan request. Will take a look.